### PR TITLE
Expansion of the Python setup on MacOS

### PR DESCRIPTION
**Pull request by JimmySunDance**
Took some time to rewrite the Python setup process for MacOS (and Linux?) - I have expanded on certain sections and hopefully made other areas clearer for those with less experience. I have also added sources and resources.

### CODE REVIEW

File: modules/core/Python.md [hunk 1]
```diff
@@ -1,25 +1,115 @@
...
+```
+Now, when you install something using pip, the package will instead be installed in the `<env_name>/` folder and not to your systems base environment. 
+
+To leave the virtual environment, run:
+```
+$ deactivate
+```
+
+## Using Virtualenv
+Below is outlined how to run up a virtual environment if you installed using Python3 (if you are using Python 2 then please refer to this documentation: [virtualenv](https://packaging.python.org/en/latest/key_projects/#virtualenv) but mostly it is swapping venv to virtualenv).
```
Comment (returnString) [inline modules/core/Python.md:102]: I hate that I have to ask this as a genuine question, but do we really have any projects still on py2?

**Review by returnString — APPROVED** [review]
lgtm - it's a damning indictment on the ecosystem that this much documentation is required, but awesome work!

Just left a minor comment on mentioning Python 2, but not fussed

File: modules/core/Python.md (hunk 1 above)
Comment (JimmySunDance) [reply modules/core/Python.md:102]: Not to my knowledge - but that is definitely limited.

File: modules/core/Python.md [hunk 2]
```diff
@@ -1,25 +1,115 @@
+# Foreword
+The following guide take you through the provisioning of your laptop assuming you are using macOS or Linux, for a windows guide to setting up your devise with python please try [here](https://docs.python.org/3/using/windows.html)
```
Comment (Lee-250) [inline modules/core/Python.md:2]: Typo:  'devise' to 'device'

File: modules/core/Python.md [hunk 3]
```diff
@@ -1,25 +1,115 @@
...
+```
+
+#### Specify global Python version
+```
+$ pyenv global < version number >
+```
+
+# Creating a virtual environment
+
+virtual environments are an essential tall in development. It ensures that developers are all working from the same point and makes setting up projects far easier. A virtual environment is created on top of an existing Python installation, known as the virtual environment’s “base” Python, and may optionally be isolated from the packages in the base environment, so only those explicitly installed in the virtual environment are available.
```
Comment (Lee-250) [inline modules/core/Python.md:75]: Typo: 'tall' to 'tool'

File: modules/core/Python.md (hunk 1 above)
Comment (JimmySunDance) [reply modules/core/Python.md]: I have updated this to simply talk about the ability to specify Python versions. I don't think people using this doc need to worry about Py2 vs Py3 stuff.

**Review by tf75 — APPROVED** [review]

**Pull request merged**